packages/gatsby/src/redux/actions.js

```diff
@@ -87,6 +98,62 @@ actions.createPage = (page, plugin = ``, traceId) => {
 }
 
 /**
+ * Delete a layout
+ * @param {string} layout a layout object with at least the name set
+ * @example
+ * deleteLayout(layout)
+ */
+actions.deleteLayout = (layout, plugin = ``) => {
+  return {
+    type: `DELETE_LAYOUT`,
+    payload: layout,
+  }
+}
+
+/**
+ * Create a layout.
+ * @param {Object} layout a layout object
+ * @param {string} layout.id Unique id for layout
+ * @param {string} layout.component The absolute path to the component for this layout
+ * @example
+ * createLayout({
+ *   id: `myNewLayout`,
```

Comment:
Good point, we had some discussions around this at awhile ago, I used Id because it make's the pages api a little easier, you can just pass `myNewLayout` instead of finding the path, but I like simplifying it to just the path, @KyleAMathews any objections?